### DISCUSSION TITLE
Why is there no item "Compress with password" in Finder's context menu?

### DISCUSSION THREAD

**Discussion by jmuheim** [aonez/Keka#1162] [Ideas]
I just stumbled over Keka, and after all those years where I didn't find anything comparable to good ol' days' WinRAR on Windows, I finally got something that behaves similarly through the Finder's context menu.

I'm a bit disappointed though that there is no menu item where I can specify a password for the archive-to-be-created. I think it's an interesting idea to have a default password, but wouldn't it make more sense to, instead just use the default password, offer a little dialog where the default password is already entered, and where I can overwrite it for the current archive?

<img width="598" alt="CleanShot 2022-09-16 at 10 00 20@2x" src="https://user-images.githubusercontent.com/1814983/190588250-b1234b28-afe5-4bbc-9f05-56a5a96de83c.png">

Anyway, I really love Keka, and will probably buy it through the App Store soon. 😃👍

**Comment by aonez** [comment]
Indeed I need to bring this feature to the table, just need the time. It was already in the to-do list:

https://github.com/aonez/Keka/issues/26

**Comment by moniuch** [comment]
Same here. I can only get the password UI when I invoke Keka new window (default app launch), but not from the Finder contextual menu. The saved/default passwords are not an option I would personally use, esp. when sharing an archive with someone. Then I would rather use one-off/random passwords, so I would like the password UI to popup on a per-case basis.

**Comment by robations** [comment]
Just to add to this, today, I used Keka for the first time in ages*. I selected some files in Finder and chose "Compress using Keka".

(\* meaning I can't remember how it normally operates or looks like as a UI)

I saw the options dialog and added a password, and then was fairly confused because there was no "Compress" or "OK" button to continue with compression. After going round in circles for a while I realised that the window that appears is actually a drag and drop interface, and anything compressed using the Finder extension will completely bypass any options you subsequently set.

**Comment by aonez** [comment]
This options is coming in the next release, next week.
